addons/mail/static/src/core/common/composer.js

```diff
@@ -600,4 +613,31 @@ export class Composer extends Component {
             this.threadService.markAsRead(this.props.composer.thread);
         }
     }
+
+    keepContent() {
+        if (!this.env.inChatter || !this.thread) {
+            return;
+        }
```

Comment:
the code could read better if the condition was outside of the method, before calling it